cpu/sam3/periph/uart.c

```diff
@@ -49,25 +50,30 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
     /* enable clock */
     uart_poweron(uart);
 
-    /* configure pins
-       TODO: optimize once GPIO refactoring is merged */
-    uart_config[uart].rx_port->PIO_PDR = (1 << uart_config[uart].rx_pin);
-    uart_config[uart].rx_port->PIO_ABSR &= ~(1 << uart_config[uart].rx_pin);
-    uart_config[uart].tx_port->PIO_ABSR |=  (uart_config[uart].mux <<
-                                             uart_config[uart].rx_pin);
-    uart_config[uart].tx_port->PIO_PDR = (1 << uart_config[uart].tx_pin);
-    uart_config[uart].tx_port->PIO_ABSR &= ~(1 << uart_config[uart].tx_pin);
-    uart_config[uart].tx_port->PIO_ABSR |=  (uart_config[uart].mux <<
-                                             uart_config[uart].tx_pin);
+    /* reset configuration */
+    dev->UART_CR = 0;
+    dev->UART_IDR = 0x0000ffff;
+
+    /* configure pins */
+    gpio_init(uart_config[uart].tx_pin, GPIO_OUT);
+    gpio_init_mux(uart_config[uart].tx_pin, uart_config[uart].mux);
+    if (rx_cb) {
+        gpio_init(uart_config[uart].rx_pin, GPIO_IN);
```

Comment:
I can't explain it either. I also tried with `GPIO_IN_PU` and it also works. There's nothing about this in the datasheet.